CONTRIBUTING.rst

```diff
@@ -105,19 +142,165 @@ This infrastructure is thoroughly described in this `wiki_section <https://githu
+Unless there is a very good reason, we discourage the use of pylint disable statements.
 
 Extra documentation
 -------------------
 
 In case you're having issues with git and working through a branch feature, you might want to have a look at this old'ish `wiki <https://github.com/dmwm/WMCore/wiki/Developing-against-WMCore>`_ in our WMCore wiki documentation.
 In addition to that, we've also compiled a long list of important git `commands <https://github.com/dmwm/WMCore/wiki/git-commands>`_. If none of those work for you, google and stackoverflow will be your best friend.
 
+
+Structural and in-depth documentation
+============
+
+WMCore structure
+-------------------
+
+When developing utilitarian libraries that do **not** depend on any of the
+WMCore libraries, create it under::
+
+    src/python/Utils/
+
+this package can be easily shared with external projects, so please ensure
+it's well covered by unit tests and that its documentation (docstrings) are
+as clean and clear as possible.
+
+Core libraries, which can be shared among the several WMCore services, should
+be implemented under::
+
+    src/python/WMCore/
+
+some of those are also shared with external projects.
+
+When developing a new WMAgent component - which inherits from the ``Harness``
+module, please use the following structure::
+
+    src/python/WMComponent/<component name>/<component name.py>
+
+and if it requires any specific config file, use::
+
+    src/python/WMComponent/<component name>/DefaultConfig.py
+
+WMCore/bin/wmcore-new-config is a config file aggregator that takes
+as input directories roots and aggregates the config files. This enables
+operators to generate one config file and edit it as they see fit. Beware that
+if you import a DefaultConfig.py file in your DefaultConfig.py file this
+can give errors when generating this file as it would overwrite existing
+values.
+
+If components contain parameters, those should be defined in the WMAgent default configuration/template file::
+
+    etc/WMAgentConfig.py
+
+
+On what concerns the tests, the module::
+
+    src/python/WMQuality/TestInit.py
+
+contains a set of methods often used in testing for performing mundane tasks,
+such as setting up database connections, deploying/cleaning database schema, etc.
+It is recommended that you use them to facilitate maintainability.
+
+To facilitate using methods from the TestInit class for loading schemas, put
+create statements in a ``...<backend>/Create.py`` method, following a similar
+structure as can be found in the Create.py methods under ``./src/python/WMCore/Agent/Database/``.
+Backend has either the value ``Oracle`` or ``MySQL``.
+
+If you are creating a new WMCore package, or making changes for an external
+project, verify whether your package - or an upper package - is listed under
+the list of packages for each service, in the file::
+
+    setup_dependencies.py
```

Comment:
And also, maybe it is not a bad idea for all those paths to be links pointing to the actual place in the master branch of the code repository. Something like:
[src/python/WMQuality/TestInit.py](https://github.com/dmwm/WMCore/blob/master/src/python/WMQuality/TestInit.py)